src/component/LoginForm.js

```diff
@@ -9,19 +10,31 @@ class LoginForm extends React.Component {
       password: ''
    };
 
+   componentDidMount() {
+
+      let arr = [];
+      arr = JSON.parse(localStorage.getItem('user'));
+
+      if (arr != null) {
+         this.props.history.push('/homepage');
+      }
+   }
+
    authenticateUser = () => {
       const temp = JSON.parse(localStorage.getItem('user'));
```

Comment:
@RajanTank all of these are diff. tasks.

1. Read from localStorage.
2. parse JSON

There should be common methods extracted so you don't repeat the same code, and so less bugs! :sunglasses: